### PR TITLE
core/api: new endpoint "/mockhsm/create-block-key"

### PR DESCRIPTION
**Pull request by boymanjor**
Previously, creating a block signing key was executed by `corectl`
through a database connection. This endpoint helps facilitate our
goal of forcing `corectl` functionality through our ACLs.

### CODE REVIEW

**Comment by boymanjor** [comment]
PTAL

**Comment by kr** [comment]
:eyes:

File: core/run.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ func RateLimit(keyFn func(*http.Request) string, burst, perSecond int) RunOption
 // used for Chain Core Developer Edition to expose the configuration UI
 // in the dashboard. API authentication still applies to an unconfigured
 // Chain Core.
-func RunUnconfigured(ctx context.Context, db pg.DB, raftDB *raft.Service, opts ...RunOption) *API {
+func RunUnconfigured(ctx context.Context, db pg.DB, raftDB *raft.Service, opts []RunOption) *API {
```
Comment (kr) [inline core/run.go]: I suggest leaving this as `...RunOption` so it matches Run.

File: cmd/corectl/no_mockhsm.go [hunk 2]
```diff
@@ -4,6 +4,6 @@ package main
 
 import "chain/database/sql"
 
-func createBlockKeyPair(db *sql.DB, args []string) {
-	fatalln("error: create-block-keypair disabled in prod build")
+func createBlockKeyPair(*sql.DB, []string) {
+	fatalln("error: create-block-keypair disabled in this build")
```
Comment (kr) [inline cmd/corectl/no_mockhsm.go]: Now that this is protected by cored, I think we can get rid of corectl's no_mockhsm build tag.

**Comment by boymanjor** [comment]
Updated, PTAL

**Pull request merged**